### PR TITLE
WIP attached types

### PR DESCRIPTION
**Pull request by aep**

### CODE REVIEW

File: modules/vec/src/main.zz [hunk 1]
```diff
@@ -19,5 +29,22 @@ export fn main() int {
     v = l.items[1];
     log::info(">%.*s<", v.size, v.mem);
 
+
+
+
+
+
+    new[vec::item(self) == typeof(A), +0] l2 = vec::make_with_pool(&p);
```
Comment (jwerle) [inline modules/vec/src/main.zz:37]: nice

File: tests/mustpass/sizeof/src/main.zz [hunk 2]
```diff
@@ -0,0 +1,6 @@
+using log;
+
+export fn main() int {
```
Comment (jwerle) [inline tests/mustpass/sizeof/src/main.zz:3]: Return type no longer needs a preceding `->`?

Comment (aep) [reply tests/mustpass/sizeof/src/main.zz:3]: correct, see https://github.com/zetzit/zz/commit/daa43997f7275b8cbfe65ab6a1b1c3a8be11a665

Comment (jwerle) [reply tests/mustpass/sizeof/src/main.zz:3]: somehow missed that!

**Pull request merged**